events/emitter.go

```diff
@@ -0,0 +1,62 @@
+package events
+
+import (
+	"context"
+	"sync"
+
+	"github.com/containerd/containerd/api/types/event"
+	"github.com/containerd/containerd/namespaces"
+	goevents "github.com/docker/go-events"
+)
+
+const (
+	EventVersion = "v1"
+)
+
+type Emitter struct {
+	sinks       map[string]*eventSink
+	broadcaster *goevents.Broadcaster
+	m           *sync.Mutex
+}
+
+func NewEmitter() *Emitter {
+	return &Emitter{
+		sinks:       make(map[string]*eventSink),
+		broadcaster: goevents.NewBroadcaster(),
```

Comment:
The broadcaster needs to be closed when you're done using it. Otherwise, it leaks a goroutine.

We use this wrapper in swarmkit, which may be helpful as an example: https://github.com/docker/swarmkit/blob/master/watch/watch.go